### PR TITLE
refactor: Update CSS form styles and structure to match React implementation

### PR DESCRIPTION
**Pull request by stephanerangaya**
This PR updates forms elements CSS styles so they match their React component.

|                  | Visual | Code |
|------------------|--------|------|
| Breaking Changes | Yes    | Yes   |

### Core
- `$wdc-color-alert` changed from cantaloupe 400 to 500.
- `$wdc-color-alert-light` changed from cantaloupe 200 to 400
- `$wdc-color-alert-dark` changed from cantaloupe 500 to 600

### Forms

All inputs are now inside a class `wdc-form-TYPE_OF_INPUT` so we can better handle accessibility and errors.

The class `wdc-form-inline-labels` that allow the label to be placed to the left of the form element is now `wdc-form-label-position-left`

### Label

The class didn't change but the style did.

### Hint

We removed `wdc-form-text`, `wdc-form-error-message` and `wdc-form-alert-message` and now using `wdc-form-hint-message`. We no longer differentiate the type of hint for errors and alerts so the styles and the classes have been updated accordingly.

### Checkbox
Both the visual and the code have changed.
```html
<div className="wdc-form-checkbox">
  <input type="checkbox" id="checkbox" />
  <label htmlFor="checkbox">Checkbox option</label>
</div>
```

### Radio
Both the visual and the code have changed.
```html
<div className="wdc-form-radio">
  <input type="radio" id="radio" />
  <label htmlFor="radio">E-mail</label>
</div>
```
### Select

Both the visual and the code have changed.
```html
<div className="wdc-form-select">
  <select id="select">
    <option label="E-mail">E-mail</option>
    <option label="Phone">Phone</option>
    <option label="Fax (disabled)" disabled={true}>
      Fax (disabled)
    </option>
    <option label="Mail">Mail</option>
  </select>
</div>
```

### Text Area
Both the visual and the code have changed.
```html
<div className="wdc-form-textarea">
  <textarea id="textarea" />
</div>
```
### Text Input
Both the visual and the code have changed.
```html
<div className="wdc-form-textinput">
  <input id="input" type="text" />
</div>
```

### Squash and Merge Message Proposal
refactor(Forms): Update CSS styles to match React implementation (#112)

### CODE REVIEW

File: modules/common/css/lib/errors.scss [hunk 1]
```diff
@@ -1,149 +1,189 @@
 // Errors
-@mixin wdc-form-validation($_borderColor) {
-  border-color: $_borderColor;
+@mixin wdc-form-validation($_type) {
 
-  &.wdc-form-checkbox:not(.wdc-form-checked):not(:checked) + .wdc-form-label:before {
+  $_borderColor: #000 !default;
```
Comment (mannycarrera4) [inline modules/common/css/lib/errors.scss]: should we use blackpepper for black?

File: modules/common/css/lib/errors.scss [hunk 2]
```diff
@@ -1,149 +1,189 @@
...
+      @if ($_type=='alert') {
+        border-color: $wdc-color-alert-dark;
+        box-shadow: inset 0 0 0 2px $_borderColor;
+      }
     }
   }
 
-  &.wdc-form-focus,
-  &:focus {
+  &.wdc-form-focus:not([disabled]):not(.wdc-form-disabled),
```
Comment (mannycarrera4) [inline modules/common/css/lib/errors.scss]: in button css update, we removed the `wdc-...-disabled` class name and just used the native disabled attribute. Should we do the same for forms?

File: modules/common/css/lib/forms.scss [hunk 3]
```diff
@@ -1,20 +1,24 @@
 // Styles common to elements with text boxes like text inputs and selects
 @mixin wdc-form-common-box() {
   @include wdc-type();
-  @include wdc-type-body-2();
+  @include wdc-type-body();
   background-color: $wdc-color-input-background;
   display: block;
   border: 1px solid $wdc-color-input-border;
-  border-radius: 3px;
+  border-radius: 4px;
```
Comment (mannycarrera4) [inline modules/common/css/lib/forms.scss]: `$wdc-spacing-xxxs`

File: modules/common/css/lib/forms.scss [hunk 4]
```diff
@@ -1,20 +1,24 @@
...
   min-width: $wdc-form-field-min-width;
   line-height: $wdc-spacing-s - 2;
   padding: $wdc-spacing-xxs;
   box-sizing: border-box;
   transition: 0.2s border-color, 0.2s box-shadow;
+  height: 40px;
```
Comment (mannycarrera4) [inline modules/common/css/lib/forms.scss]: `$wdc-spacing-xl`

File: modules/common/css/lib/forms.scss [hunk 5]
```diff
@@ -1,20 +1,24 @@
...
 
-  &.wdc-form-hover:not([disabled]):not(.wdc-form-disabled),
-  &:hover:not([disabled]):not(.wdc-form-disabled) {
+  &.wdc-form-focus:not([disabled]):not(.wdc-form-disabled),
+  &:focus:not([disabled]):not(.wdc-form-disabled) {
     border-color: $wdc-color-input-hover-border;
+    border-color: #0875e1;
```
Comment (mannycarrera4) [inline modules/common/css/lib/forms.scss]: this is probably blueberry something ?

File: modules/form-field/css/lib/form-field.scss [hunk 6]
```diff
@@ -5,6 +5,8 @@
 
   .wdc-form-field {
     position: relative;
+    align-items: center;
```
Comment (mannycarrera4) [inline modules/form-field/css/lib/form-field.scss]: Does this class need `display: flex` for the align items to work with flex box?

File: modules/common/css/lib/forms.scss (hunk 5 above)
Comment (stephanerangaya) [reply modules/common/css/lib/forms.scss]: $wdc-color-blueberry-400

File: modules/common/css/lib/errors.scss (hunk 2 above)
Comment (anicholls) [reply modules/common/css/lib/errors.scss]: @mannycarrera4 we didn't entirely remove the class from the scss, we just changed the stories to use the attribute instead of the class so they weren't focusable. Let's leave this as is.

File: modules/checkbox/css/README.md [hunk 7]
```diff
@@ -37,4 +40,4 @@ made fully accessible.
 
 ## Labels
 
-Render a checkbox inside a [canvas-kit-form-field](../../form-field/css) to add a label.
+Render a select inside a [canvas-kit-form-field](../../form-field/css) to add a label.
```
Comment (NicholasBoll) [inline modules/checkbox/css/README.md]: Was this change intentional? I think it is supposed to be "checkbox", right?

File: modules/form-field/css/package.json [hunk 8]
```diff
@@ -20,6 +20,13 @@
     "@workday/canvas-kit-css-core": "^3.0.0-alpha.6",
     "@workday/canvas-system-icons-web": "^1.0.1"
   },
+  "devDependencies": {
+    "@workday/canvas-kit-css-checkbox": "^3.0.0-alpha.0",
```
Comment (NicholasBoll) [inline modules/form-field/css/package.json]: I'm guessing these `devDependencies` are for the Storybook stories?

File: modules/form-field/css/lib/form-field.scss [hunk 9]
```diff
@@ -5,6 +5,9 @@
 
   .wdc-form-field {
     position: relative;
+    min-height: 32px;
```
Comment (NicholasBoll) [inline modules/form-field/css/lib/form-field.scss]: There are a lot of magic numbers like this throughout our CSS code. Some CSS has addition between spacing values. Other values don't have and spacing match. Is there rationale behind when we choose values?

File: modules/form-field/css/stories/stories_TextArea.tsx [hunk 10]
```diff
@@ -29,7 +31,9 @@ storiesOf('CSS/Form Field/TextArea/Top Label', module)
             Label
           </label>
           <div className="wdc-form-field">
-            <textarea id="textarea" className="wdc-form-textarea" placeholder="Placeholder" />
+            <div className="wdc-form-textarea">
+              <textarea id="textarea" placeholder="Placeholder" />
```
Comment (NicholasBoll) [inline modules/form-field/css/stories/stories_TextArea.tsx]: Multiple examples use the same `id`. Should we use different ids?

File: modules/form-field/css/stories/stories_TextArea.tsx [hunk 11]
```diff
@@ -61,11 +67,9 @@ storiesOf('CSS/Form Field/TextArea/Top Label', module)
             Label
           </label>
           <div className="wdc-form-field">
-            <textarea
-              className="wdc-form-textarea wdc-form-disabled"
-              placeholder="Placeholder"
-              disabled={true}
-            />
+            <div className="wdc-form-textarea wdc-form-disabled">
+              <textarea placeholder="Placeholder" disabled={true} />
```
Comment (NicholasBoll) [inline modules/form-field/css/stories/stories_TextArea.tsx]: This example has a `label` with an `htmlFor="textarea"`, but there is no `id` here

File: modules/form-field/css/stories/stories_TextArea.tsx [hunk 12]
```diff
@@ -162,11 +176,13 @@ storiesOf('CSS/Form Field/TextArea/Left Label', module)
...
           </label>
           <div className="wdc-form-field">
-            <textarea
-              className="wdc-form-textarea wdc-form-disabled"
-              placeholder="Placeholder"
-              disabled={true}
-            />
+            <div className="wdc-form-textarea">
+              <textarea
+                className="wdc-form-textarea wdc-form-disabled"
```
Comment (NicholasBoll) [inline modules/form-field/css/stories/stories_TextArea.tsx]: There is a `label` with an `htmlFor`, but no `id` on the `textarea`

File: modules/form-field/css/stories/stories_TextInput.tsx [hunk 13]
```diff
@@ -67,12 +63,9 @@ storiesOf('CSS/Form Field/Text Input/Top Label', module)
...
           </label>
           <div className="wdc-form-field">
-            <input
-              type="text"
-              className="wdc-form-textinput wdc-form-disabled"
-              placeholder="Placeholder"
-              disabled={true}
-            />
+            <div className="wdc-form-textinput wdc-form-disabled">
+              <input type="text" placeholder="Placeholder" disabled={true} />
```
Comment (NicholasBoll) [inline modules/form-field/css/stories/stories_TextInput.tsx]: There is no `id` on this `input`. The label has an `htmlFor` of `input`

File: modules/form-field/css/stories/stories_TextInput.tsx [hunk 14]
```diff
@@ -175,12 +168,9 @@ storiesOf('CSS/Form Field/Text Input/Left Label', module)
...
```
Comment (NicholasBoll) [inline modules/form-field/css/stories/stories_TextInput.tsx]: There is no `id`. The label has an `htmlFor="input"`

File: modules/form-field/react/stories/stories_TextInput.tsx [hunk 15]
```diff
@@ -14,139 +14,175 @@ const hintId = 'error-desc-id';
...
+    </div>
   ));
 
 storiesOf('Form Field/Text Input/Left Label', module)
   .addDecorator(withReadme(README))
   .add('Default', () => (
-    <FormField labelPosition={FormField.LabelPosition.Left} label="Label" inputId="input-plain">
-      {controlComponent(<TextInput />)}
-    </FormField>
+    <div className="story">
```
Comment (NicholasBoll) [inline modules/form-field/react/stories/stories_TextInput.tsx]: What does this wrapper div do?

File: modules/radio/css/README.md [hunk 16]
```diff
@@ -33,4 +36,4 @@ made fully accessible.
 
 ## Labels
 
-Render a radio inside a [canvas-kit-form-field](../../form-field/css) to add a label.
+Render a select inside a [canvas-kit-form-field](../../form-field/css) to add a label.
```
Comment (NicholasBoll) [inline modules/radio/css/README.md]: Is this supposed to be "radio"?

Comment (stephanerangaya) [reply modules/radio/css/README.md]: yes

File: modules/checkbox/css/README.md (hunk 7 above)
Comment (stephanerangaya) [reply modules/checkbox/css/README.md]: yes

File: modules/core/css/lib/variables/colors.scss [hunk 17]
```diff
@@ -4,8 +4,8 @@
 $wdc-color-error: $wdc-color-cinnamon-500;
 $wdc-color-error-light: $wdc-color-cinnamon-200;
 $wdc-color-error-dark: $wdc-color-cinnamon-600;
-$wdc-color-alert: $wdc-color-cantaloupe-400;
-$wdc-color-alert-light: $wdc-color-cantaloupe-200;
-$wdc-color-alert-dark: $wdc-color-cantaloupe-500;
+$wdc-color-alert: $wdc-color-cantaloupe-500;
```
Comment (anicholls) [inline modules/core/css/lib/variables/colors.scss]: These should be documented as breaking changes as well

File: modules/form-field/css/README.md [hunk 18]
```diff
@@ -125,18 +125,18 @@ Error styling is available as both classes and mixins. Using the class is prefer
 ### Form Fields
 
 Use `.wdc-form-field-error` for errors and `.wdc-form-field-alert` for alerts. Applying error and
-alert styling will display an icon on the left next to the input.
+alert styling will display an icon on the right inside the input.
 
 **Messages**  
-Add messages to errors and alerts by wrapping your message with `.wdc-form-error-message` or
-`.wdc-form-alert-message`. Using `strong` will bolden text with the respective error/alert color.
+Add messages to errors and alerts by wrapping your message with `.wdc-form-hint-message` or
```
Comment (anicholls) [inline modules/form-field/css/README.md]: Need to call this out as a breaking change

File: modules/form-field/css/README.md [hunk 19]
```diff
@@ -48,10 +48,10 @@ inline icons. Process your SASS through PostCSS once it has been compiled to CSS
 ## Inline Labels
 
 Form labels are rendered above the fields by default. You can enable inline labels (side by side),
-by applying the `.wdc-form-inline-labels` class to `.wdc-form`.
+by applying the `.wdc-form-label-position-left` class to `.wdc-form`.
 
 ```html
-<div class="wdc-form wdc-form-inline-labels">
+<div class="wdc-form wdc-form-label-position-left">
```
Comment (anicholls) [inline modules/form-field/css/README.md]: Need to call this out as a breaking change

File: modules/form-field/css/lib/form-field.scss (hunk 9 above)
Comment (anicholls) [reply modules/form-field/css/lib/form-field.scss]: +1. We should always use the spacing values if possible, and if we're doing additions, we should denote the addition happening, rather than just using the result.

File: modules/form-field/css/package.json (hunk 8 above)
Comment (anicholls) [reply modules/form-field/css/package.json]: This will probably get bumped when we version, but all of these should be `alpha.6`

File: modules/form-field/react/stories/stories_Select.tsx [hunk 20]
```diff
@@ -107,7 +107,6 @@ storiesOf('Form Field/Select/Top Label', module)
   ));
 
 storiesOf('Form Field/Select/Left Label', module)
-  .addDecorator(SectionDecorator('Select'))
```
Comment (anicholls) [inline modules/form-field/react/stories/stories_Select.tsx]: How come this was removed for the left label and not the top one?

File: modules/form-field/react/stories/stories_TextArea.tsx [hunk 21]
```diff
@@ -14,143 +14,179 @@ const hintId = 'error-desc-id';
 storiesOf('Form Field/TextArea/Top Label', module)
   .addDecorator(withReadme(README))
   .add('Default', () => (
-    <FormField label="Label" inputId="textarea-plain">
-      {controlComponent(<TextArea />)}
-    </FormField>
+    <div className="story">
```
Comment (anicholls) [inline modules/form-field/react/stories/stories_TextArea.tsx]: You could update the `sectionDecorator` helper to remove the title (since we don't use those anymore now that we render the stories on the design site) and use that to avoid all this extra markup.

File: modules/form-field/react/stories/stories_TextInput.tsx (hunk 15 above)
Comment (anicholls) [reply modules/form-field/react/stories/stories_TextInput.tsx]: +1 - why do we need this change? It's adding a lot of lines changed. If we need it, let's refactor `sectionDecorator` so we can use it instead.

File: modules/radio/css/lib/radio.scss [hunk 22]
```diff
@@ -1,80 +1,14 @@
...
-    &:focus {
-      @include wdc-form-radio-checked-focus();
-    }
-  }
+  @include wdc-form-selection-control();
 
-  &:focus {
-    @include wdc-form-radio-focus();
+  input+label::before {
+    border-radius: 18px;
```
Comment (anicholls) [inline modules/radio/css/lib/radio.scss]: Let's just make this 100% so we don't have to worry about the size changing.

File: modules/common/css/lib/forms.scss [hunk 23]
```diff
@@ -32,46 +36,16 @@
...
-  & + .wdc-form-label {
-    @include wdc-type-body-2();
-    display: inline-block;
-    margin: 0 0 0 $wdc-spacing-xxs;
-    top: -4px;
-  }
 
   &:disabled,
   &.wdc-form-disabled {
+    background: $wdc-color-soap-100;
```
Comment (anicholls) [inline modules/common/css/lib/forms.scss]: Need to change the cursor here

File: modules/form-field/css/stories/stories_Checkbox.tsx [hunk 24]
```diff
@@ -1,22 +1,22 @@
 import React from 'react';
 import {storiesOf} from '@storybook/react';
 import withReadme from 'storybook-readme/with-readme';
-import README from '../README.md';
+import README from '../../../checkbox/css/README.md';
 import '../index.scss';
 import '@workday/canvas-kit-css-checkbox/index.scss';
 
-storiesOf('CSS/Form Field/Checkbox/Top Label/Checkbox', module)
+storiesOf('CSS/Form Field/Checkbox/Top Label', module)
```
Comment (anicholls) [inline modules/form-field/css/stories/stories_Checkbox.tsx]: Missing `/Checkbox` on the end of this path

File: modules/form-field/css/lib/form-field.scss [hunk 25]
```diff
@@ -67,13 +73,34 @@
...
+      // Case where there's no label on the left
+      flex-grow: 0;
+
+      .wdc-form-checkbox:first-child,
+      .wdc-form-radio:first-child {
+        margin-top: $wdc-spacing-xxxs;
+
+        &.wdc-form-alert,
+        &.wdc-form-error {
+          margin-top: 0;
```
Comment (anicholls) [inline modules/form-field/css/lib/form-field.scss]: The padding here is incorrect in both CSS & React. I know you just matched what was in React, but can you double check it with the specs and update both?

File: modules/core/css/lib/mixins/type.scss [hunk 26]
```diff
@@ -2,8 +2,6 @@
 
 @mixin wdc-type() {
   font-family: $wdc-font-family;
-  -webkit-font-smoothing: antialiased;
```
[the remark this answers is not in the document]
Comment (stephanerangaya) [reply modules/core/css/lib/mixins/type.scss]: This was removed to match the react component, will revert.

File: modules/form-field/css/package.json (hunk 8 above)
Comment (stephanerangaya) [reply modules/form-field/css/package.json]: I'm actually not sure what devDependencies do here?

File: modules/form-field/css/stories/stories_TextArea.tsx (hunk 10 above)
Comment (stephanerangaya) [reply modules/form-field/css/stories/stories_TextArea.tsx]: I think it's fine, because they're not displayed simultaneously

**Comment by NicholasBoll** [comment]
I updated the PR description to use `html` for the code block for better syntax highlighting ```html

File: modules/common/css/lib/variables.scss [hunk 27]
```diff
@@ -1,6 +1,8 @@
-$wdc-form-error-icon-size: 20px;
+$wdc-form-error-icon-size: 24px;
 $wdc-form-error-spacing: $wdc-spacing-xxs;
 $wdc-form-label-line-height: 16px;
-$wdc-form-inline-label-width: 176px;
+$wdc-form-inline-label-width: 180px;
 $wdc-form-field-min-width: 280px;
+$wdc-form-textarea-min-height: $wdc-spacing-xxl;
+$wdc-form-selection-control-spacing: 18px;
```
Comment (anicholls) [inline modules/common/css/lib/variables.scss]: Since this is the width/height of checkboxes and radios, I would call this the size, not the spacing.

File: modules/core/css/lib/variables/colors.scss [hunk 28]
```diff
@@ -4,8 +4,9 @@
...
+$wdc-color-alert-light: $wdc-color-cantaloupe-400;
+$wdc-color-alert-dark: $wdc-color-cantaloupe-600;
 $wdc-color-shadow-light: rgba(82, 97, 115, 0.14);
 $wdc-color-shadow: rgba(82, 97, 115, 0.18);
+$wdc-color-input-focus-border: $wdc-color-blueberry-400;
```
Comment (anicholls) [inline modules/core/css/lib/variables/colors.scss]: This shouldn't get added here if you need it, it would go in design-assets. I also think it already exists as `$wdc-color-input-focus-outline`

File: modules/form-field/css/package.json (hunk 8 above)
Comment (anicholls) [reply modules/form-field/css/package.json]: Yes, they're for the stories 👍
